File: How to transform XML using XSLT, all parameters as strings. No file system/[VB]-How to transform XML using XSLT, all parameters as strings. No file system/VB/XmlTransformation/Core/ResourcesReader.vb
Imports System.Reflection
Imports System.IO

Namespace XmlTransformation

	Public Class ResourcesReader
		''' <exception cref="ArgumentException">Cannot find a resources with the specified name.</exception>
		Public Function Read(ByVal name As String) As String
			Using stream = System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(name)
				If stream Is Nothing Then
					Throw New ArgumentException("Cannot find a resources with the specified name.")
				End If

				Using reader = New StreamReader(stream)
					Return reader.ReadToEnd()
				End Using
			End Using
		End Function
	End Class
End Namespace

File: How to transform XML using XSLT, all parameters as strings. No file system/[C#]-How to transform XML using XSLT, all parameters as strings. No file system/C#/XmlTransformation/Core/ResourcesReader.cs
namespace XmlTransformation
{
	using System;
	using System.IO;
	using System.Reflection;

	public class ResourcesReader
	{
		/// <exception cref="ArgumentException">Cannot find a resources with the specified name.</exception>
		public string Read(string name)
		{
			using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(name))
			{
				if (stream == null)
				{
					throw new ArgumentException("Cannot find a resources with the specified name.");
				}

				using (var reader = new StreamReader(stream))
				{
					return reader.ReadToEnd();
				}
			}
		}
	}
}